clinica/pydra/engine_utils.py

```diff
@@ -48,19 +52,53 @@ def bids_query(keys: list) -> dict:
     return {key: bids_keys_available[key] for key in keys}
 
 
-def run(wf: Workflow) -> None:
-
+def run(wf: Workflow) -> str:
     """Execute a Pydra workflow
 
     Parameters
     ----------
         wf : Workflow
-            The workflow to execute
+
+    Returns
+    -------
+    Union[str, pydra.engine.specs.Result]
+        The result of running the Workflow
+
+    """
+    import re
+
+    try:
+        with Submitter(plugin="cf") as submitter:
+            submitter(wf)
+    except Exception as e:
+        path = re.search("\/.*\.pklz", str(e))
+        if path:
+            path = path.group(0)
+            print(read_error(path))
+        else:
+            print("Exception:", e)
+        return str(e)
```

Comment:
I would raise the error instead of printing and returning it. This way we avoid the weird return type and make sure the execution stops. WDYT?

Reply:
I ended up doing a return directly to avoid the traceback messages. Seems to be the most readable version.

Reply:
@omar-rifai I was thinking about something along those lines (I haven't tested, there are probably some mistakes to fix and some changes to make...):

```python
class WorkflowExecutionError(Exception):
    def __init__(self, message="Error running the Workflow."):
        import re
        self.message = message
        path = re.search("\/.*\.pklz", self.message)
        if path:
            self.message = read_error(path.group(0))
        super().__init__(self.message)

def read_error(path: PathLike) -> str:
    with open(path, "rb") as fp:
        err = cp.load(fp)
    return err["error message"][:-1]

try:
    with Submitter(plugin="cf") as submitter:
        submitter(wf)
except Exception as e:
    raise WorkflowExecutionError(str(e))
```

Reply:
As discussed offline, we can leave this suggestion for a release version and keep the current version for debugging.